tools/common/schema/updatetask.go

```diff
@@ -297,88 +299,109 @@ func readManifest(dirPath string) (*manifest, error) {
 	return &manifest, nil
 }
 
-// readSchemaDir returns a sorted list of subdir names that hold
-// the schema changes for versions in the range startVer < ver <= endVer
-// when endVer is empty this method returns all subdir names that are greater than startVer
-// this method has an assumption that the subdirs containing the
-// schema changes will be of the form vx.x, where x.x is the version
-// returns error when
-//  - startVer < endVer
-//  - endVer is empty and no subdirs have version >= startVer
-//  - endVer is non-empty and subdir with version == endVer is not found
-func readSchemaDir(dir string, startVer string, endVer string) ([]string, error) {
+// sortAndFilterVersions returns a sorted list of semantic versions the fall within the range
+// (startVerExcl, endVerIncl]. If endVerIncl is not specified, returns all versions > startVerExcl.
+// If endVerIncl is specified, it must be present in the list of versions.
+func sortAndFilterVersions(versions []string, startVerExcl string, endVerIncl string, logger log.Logger) ([]string, error) {
 
-	subdirs, err := os.ReadDir(dir)
+	startVersionExclusive, err := semver.ParseTolerant(startVerExcl)
 	if err != nil {
 		return nil, err
 	}
 
-	hasEndVer := len(endVer) > 0
+	var endVersionInclusive *semver.Version
+	if len(endVerIncl) > 0 {
+		evi, err := semver.ParseTolerant(endVerIncl)
+		if err != nil {
+			return nil, err
+		}
+		endVersionInclusive = &evi
 
-	if hasEndVer && cmpVersion(startVer, endVer) > 0 {
-		return nil, fmt.Errorf("startVer (%v) must be less than or equal to endVer (%v)", startVer, endVer)
+		if startVersionExclusive.Compare(*endVersionInclusive) >= 0 {
```

Comment:
needs to check when there is no schema update, this won't fail any pipeline (in case they call this using same start/end ver).

Reply:
Done